be/src/olap/tablet_manager.cpp

```diff
@@ -693,12 +694,16 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(
         ReadLock rlock(tablets_shard.lock.get());
         for (const auto& tablet_map : tablets_shard.tablet_map) {
             for (const TabletSharedPtr& tablet_ptr : tablet_map.second.table_arr) {
-                std::vector<TTabletId>::iterator it_tablet =
-                        find(tablet_submitted_compaction.begin(), tablet_submitted_compaction.end(),
-                             tablet_ptr->tablet_id());
-                if (it_tablet != tablet_submitted_compaction.end()) {
+                std::set<TTabletId>::iterator it_tablet =
+                        (*tablet_submitted_base_compaction).find(tablet_ptr->tablet_id());
+                if (it_tablet != (*tablet_submitted_base_compaction).end()) {
                     continue;
                 }
+                it_tablet = (*tablet_submitted_cumulative_compaction).find(tablet_ptr->tablet_id());
+                if (it_tablet != (*tablet_submitted_cumulative_compaction).end()) {
+                    continue;
+                }
+
```

Comment:
You can use `ContainsKey` in be/src/gutil/map-util.h instead to simplify code.

Reply:
> You can use `ContainsKey` in be/src/gutil/map-util.h instead to simplify code.

We need to judge whether an element exists in `set`rather than `map`.